### ISSUE TITLE
Release 4.8.0 - Beta 1 - E2E UX tests - GCP integration

### ISSUE THREAD

**Issue by davidjiglesias** [wazuh/wazuh#21769]
## End-to-End (E2E) Testing Guideline

- **Documentation:** Always consult the development documentation for the current stage tag at [this link](https://documentation-dev.wazuh.com/v4.8.0-beta1/index.html). Be careful because some of the description steps might refer to a current version in production, always navigate using the current development documention for the stage under test. Also, visit the following [pre-release package guide](https://docs.google.com/document/d/1fW8koy8P8OMvcxzIOXWGDw9qmKlju-XuRHD-Bdv3uP4/edit) to understand how to modify certain links and urls for the correct testing of the development packages.
- **Test Requirements:** Ensure your test comprehensively includes a full stack and agent/s deployment as per the Deployment requirements, detailing the machine OS, installed version, and revision.
- **Deployment Options:** While deployments can be local (using VMs, Vagrant, etc) or on the aws-dev account, opt for local deployments when feasible. For AWS access, coordinate with the CICD team through [this link](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **External Accounts:** If tests require third-party accounts (e.g., GitHub, Azure, AWS, GCP), request the necessary access through the CICD team [here](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **Alerts:** Every test should generate a minimum of one end-to-end alert, from the agent to the dashboard, irrespective of test type.
- **Multi-node Testing:** For multi-node wazuh-manager tests, ensure agents are connected to both workers and the master node.
- **Package Verification:** Use the pre-release package that matches the current TAG you're testing. Confirm its version and revision.
- **Filebeat Errors:** If you encounter errors with Filebeat during testing, refer to [this Slack discussion](https://wazuh-team.slack.com/archives/C03BDG0K6JC/p1672168163537809) for insights and resolutions.
- **Known Issues:** Familiarize yourself with previously reported issues in the Known Issues section. This helps in identifying already recognized errors during testing.
- **Reporting New Issues:** Any new errors discovered during testing that aren't listed under Known Issues should be reported. Assign the issue to the corresponding team (QA if unsure), add the `Release testing` objective and `Very high` priority. Communicate these to the team and QA via the c-release Slack channel.
- **Test Conduct:** It's imperative to be thorough in your testing, offering enough detail for reviewers. Incomplete tests might necessitate a redo.
- **Documentation Feedback:** Encountering documentation gaps, unclear guidelines, or anything that disrupts the testing or UX? Open an issue, especially if it's not listed under Known Issues. Please answer the feedback section, this is a mandatory step.
- **Format:** If this is your first time doing this, refer to the format (but not necessarily the content, as it may vary) of previous E2E tests, here you have an example https://github.com/wazuh/wazuh/issues/13994.
- **Status and completion:** Change the issue status within your team project accordingly. Once you finish testing and write the conclusions, move it to Pending review and notify the @wazuh/framework team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ). Beware that the reviewers might request additional information or task repetitions.
- **For reviewers:** Please move the issue to Pending final review and notify via Slack using the same thread if everything is ok, otherwise, perform an issue update with the requested changes and move it to On hold, increase the review_cycles in the  team project by one and notify the issue assignee via Slack using the same thread.

For the conclusions and the issue testing and updates, use the following legend:

**Status legend**
- 🟢 All checks passed
- 🟡 Found a known issue
- 🔴 Found a new error

## Issue delivery and completion
- **Initial delivery:** The issue's assignee must complete the testing and deliver the results by Feb 07, 2024 and notify the @wazuh/framework team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ)
- **Review:** The @wazuh/framework team will assign a reviewer and add it to the `review_assignee` field in the  project. The reviewer must then review the test steps and results. Ensure that all iteration cycles are completed by Feb 08, 2024 date (issue must be in `Pending final review` status) and notify the QA team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ).
- **Auditor:** The QA team must audit, validate the results, and close the issue by Feb 09, 2024.

## Deployment requirements

| Component | Installation | Type | OS |
|----------|--------------|------|----|
| Indexer | [Amazon Machine Images (AMI)](https://documentation-dev.wazuh.com/v4.8.0-beta1/deployment-options/amazon-machine-images/amazon-machine-images.html) | - | - |
| Server | [Amazon Machine Images (AMI)](https://documentation-dev.wazuh.com/v4.8.0-beta1/deployment-options/amazon-machine-images/amazon-machine-images.html) | - | - |
| Dashboard | [Amazon Machine Images (AMI)](https://documentation-dev.wazuh.com/v4.8.0-beta1/deployment-options/amazon-machine-images/amazon-machine-images.html) | - | - |
| Agent | [Installing Wazuh agents](https://documentation-dev.wazuh.com/v4.8.0-beta1/installation-guide/wazuh-agent/index.html) | - | CentOS 7 x86_64 |

## Test description

Configure the GCP integration in a Wazuh Manager and a Wazuh Agent.
Try both Pub/Sub and Storage integrations.
Follow the use cases from [blog post](https://wazuh.com/blog/cloud-security-posture-management/) section "Cloud security posture management simulation" and ensure the alerts are correctly displayed on the GCP dashboard. Make sure you follow the infrastructure and configuration details from the documentation below and not the blog one.

Documentation: https://documentation-dev.wazuh.com/v4.8.0-beta1/cloud-security/gcp/index.html

## Known issues

- https://github.com/wazuh/wazuh-documentation/issues/4996 
- https://github.com/wazuh/wazuh-documentation/issues/5699 
- https://github.com/wazuh/wazuh-documentation/issues/5505 
- https://github.com/wazuh/wazuh/issues/15615 

## Conclusions

Summarize the errors detected (Known Issues included). Illustrate using the table below. **REMOVE CURRENT EXAMPLES**:

| **Status**     | **Test**    | **Failure type**    | **Notes**      |
|----------------|-------------|---------------------|----------------|
| 🟢  | Configuring GCP credentials |  |  |
| 🟢  | Google Pub/Sub integration - Wazuh manager |  |  |
| 🟡 |  Bucket storage test |  | This happened because I could not create buckets since billing was disabled for the project in question |
| 🟢 | GCP integration on Wazuh agent | | |
| 🟡 | Cloud Security Posture Management ([BlogPost](https://wazuh.com/blog/cloud-security-posture-management/)) |  | Known issue from previous test: https://github.com/wazuh/wazuh/issues/21538 |

## Feedback

We value your feedback. Please provide insights on your testing experience.

- Was the testing guideline clear? Were there any ambiguities?
  - **Yes it was clear**
- Did you face any challenges not covered by the guideline?
  - **No challenges as the CICD team provided the requested permissions required **
- Suggestions for improvement:
  - **None at the time**

## Reviewers validation

The criteria for completing this task is based on the validation of the conclusions and the test results by all reviewers.

All the checkboxes below must be marked in order to close this issue.
- [x] @davidjiglesias
- [x] @wazuh/framework

**Comment by Is3x** [comment]
### Environment deployment

#### Cloud VMs

- https://github.com/wazuh/internal-devel-requests/issues/753

#### Wazuh server EC2 instance details:
```console
[wazuh-user@wazuh-server ~]$  cat /etc/os-release
NAME="Amazon Linux"
VERSION="2"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
HOME_URL="https://amazonlinux.com/"
SUPPORT_END="2025-06-30"
[wazuh-user@wazuh-server ~]$

```

#### Wazuh agent EC2 instance details:
```console
[centos@ip-172-31-38-244 ~]$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

[centos@ip-172-31-38-244 ~]$
```

**Comment by Is3x** [comment]
## GCP integration

The CICD team fulfilled my request for the necessary permissions necessary for this test.

### :green_circle: Configuring GCP credentials  

- Successfully created a **custom role**.
- Successfully created a service account and added the necessary roles: **Pub/Sub Publisher**, **Pub/Sub Subscriber** and **Storage Legacy Bucket Writer**. 
- Successfully applied the selected roles.
- Successfully created a private key and used the credential file for authentication by saving the key under `/var/ossec/wodles/gcloud/credentials.json`. 

### :green_circle: Google Pub/Sub integration - Wazuh manager

- Created a topic "Beta1_E2E_Testing4.8.0 " and added a subscription following the [Configuring Google Cloud Pub/Sub ](https://documentation-dev.wazuh.com/v4.8.0-beta1/cloud-security/gcp/prerequisites/pubsub.html)instructions. 

- Configured the `gcp-pubsub` integration on the Wazuh manager.

<details><summary> gcp-pubsub configuration </summary>

```
<gcp-pubsub>
     <pull_on_start>yes</pull_on_start>
     <interval>1m</interval>
     <project_id>beta-1-e2e-testing</project_id>
     <subscription_name>Beta1-E2E-Sub</subscription_name>
     <credentials_file>/var/ossec/wodles/gcloud/credentials.json</credential_file> 
</gcp-pubsub>
``` 
</details>

- Created a sink to monitor all the audit logs following the [Audited resources](https://documentation-dev.wazuh.com/v4.8.0-beta1/cloud-security/gcp/supported-services/cloud_audit_logs.html) documentation:
```
logName=~("projects/.*/logs/cloudaudit.googleapis.com%2F(activity|data_access|system_event|policy)")
```

And received the sent alert: 

![ttt](https://github.com/wazuh/wazuh/assets/85684507/ddd52e7c-2243-4426-b296-f0f8d9fb9a86)

Cross checked this with the alert from the console and it matches.

![tttt](https://github.com/wazuh/wazuh/assets/85684507/0a73623b-aa53-4bfb-b1fd-b32188308c4d)

**Comment by Is3x** [comment]
### :green_circle: GCP integration on Wazuh agent 

#### Configuring Wazuh agent
I followed the steps for installing a Wazuh agent on linux with the use of the yum repository considering the agent OS is CentOS.
I confirmed that the Agent was registered at the manager as could be seen below from the dashboard:
![image](https://github.com/wazuh/wazuh/assets/85684507/310686b1-523d-47dd-9002-6345aee24e40)

#### Installing Prerequisite dependencies

I was able to install all the prerequisite dependencies except one. the setuptools==68.0.0. The below error was thrown when i attempted to install this:

```console
ERROR: Could not find a version that satisfies the requirement setuptools==68.0.0 (from versions: 0.6b1, 0.6b2, 0.6b3, 0.6b4, 0.6rc1, 0.6rc2, 0.6rc3, 0.6rc4, 0.6rc5, 0.6rc6, 0.6rc7, 0.6rc8, 0.6rc9, 0.6rc10, 0.6rc11, 0.7.2, 0.7.3, 0.7.4, 0.7.5, 0.7.6, 0.7.7, 0.7.8, 0.8, 0.9, 0.9.1, 0.9.2, 0.9.3, 0.9.4, 0.9.5, 0.9.6, 0.9.7, 0.9.8, 1.0, 1.1, 1.1.1, 1.1.2, 1.1.3, 1.1.4, 1.1.5, 1.1.6, 1.1.7, 1.2, 1.3, 1.3.1, 1.3.2, 1.4, 1.4.1, 1.4.2, 2.0, 2.0.1, 2.0.2, 2.1, 2.1.1, 2.1.2, 2.2, 3.0, 3.0.1, 3.0.2, 3.1, 3.2, 3.3, 3.4, 3.4.1, 3.4.2, 3.4.3, 3.4.4, 3.5, 3.5.1, 3.5.2, 3.6, 3.7, 3.7.1, 3.8, 3.8.1, 4.0, 4.0.1, 5.0, 5.0.1, 5.0.2, 5.1, 5.2, 5.3, 5.4, 5.4.1, 5.4.2, 5.5, 5.5.1, 5.6, 5.7, 5.8, 6.0.1, 6.0.2, 6.1, 7.0, 8.0, 8.0.1, 8.0.2, 8.0.3, 8.0.4, 8.1, 8.2, 8.2.1, 8.3, 9.0, 9.0.1, 9.1, 10.0, 10.0.1, 10.1, 10.2, 10.2.1, 11.0, 11.1, 11.2, 11.3, 11.3.1, 12.0, 12.0.1, 12.0.2, 12.0.3, 12.0.4, 12.0.5, 12.1, 12.2, 12.3, 12.4, 13.0.1, 13.0.2, 14.0, 14.1, 14.1.1, 14.2, 14.3, 14.3.1, 15.0, 15.1, 15.2, 16.0, 17.0, 17.1, 17.1.1, 18.0, 18.0.1, 18.1, 18.2, 18.3, 18.3.1, 18.3.2, 18.4, 18.5, 18.6, 18.6.1, 18.7, 18.7.1, 18.8, 18.8.1, 19.0, 19.1, 19.1.1, 19.2, 19.3, 19.4, 19.4.1, 19.5, 19.6, 19.6.1, 19.6.2, 19.7, 20.0, 20.1, 20.1.1, 20.2.2, 20.3, 20.3.1, 20.4, 20.6.6, 20.6.7, 20.6.8, 20.7.0, 20.8.0, 20.8.1, 20.9.0, 20.10.1, 21.0.0, 21.1.0, 21.2.0, 21.2.1, 21.2.2, 22.0.0, 22.0.1, 22.0.2, 22.0.4, 22.0.5, 23.0.0, 23.1.0, 23.2.0, 23.2.1, 24.0.0, 24.0.1, 24.0.2, 24.0.3, 24.1.0, 24.1.1, 24.2.0, 24.2.1, 24.3.0, 24.3.1, 25.0.0, 25.0.1, 25.0.2, 25.1.0, 25.1.1, 25.1.2, 25.1.3, 25.1.4, 25.1.5, 25.1.6, 25.2.0, 25.3.0, 25.4.0, 26.0.0, 26.1.0, 26.1.1, 27.0.0, 27.1.0, 27.1.2, 27.2.0, 27.3.0, 27.3.1, 28.0.0, 28.1.0, 28.2.0, 28.3.0, 28.4.0, 28.5.0, 28.6.0, 28.6.1, 28.7.0, 28.7.1, 28.8.0, 28.8.1, 29.0.0, 29.0.1, 30.0.0, 30.1.0, 30.2.0, 30.2.1, 30.3.0, 30.4.0, 31.0.0, 31.0.1, 32.0.0, 32.1.0, 32.1.1, 32.1.2, 32.1.3, 32.2.0, 32.3.0, 32.3.1, 33.1.0, 33.1.1, 34.0.0, 34.0.1, 34.0.2, 34.0.3, 34.1.0, 34.1.1, 34.2.0, 34.3.0, 34.3.1, 34.3.2, 34.3.3, 34.4.0, 34.4.1, 35.0.0, 35.0.1, 35.0.2, 36.0.1, 36.1.0, 36.1.1, 36.2.0, 36.2.1, 36.2.2, 36.2.3, 36.2.4, 36.2.5, 36.2.6, 36.2.7, 36.3.0, 36.4.0, 36.5.0, 36.6.0, 36.6.1, 36.7.0, 36.7.1, 36.7.2, 36.8.0, 37.0.0, 38.0.0, 38.1.0, 38.2.0, 38.2.1, 38.2.3, 38.2.4, 38.2.5, 38.3.0, 38.4.0, 38.4.1, 38.5.0, 38.5.1, 38.5.2, 38.6.0, 38.6.1, 38.7.0, 39.0.0, 39.0.1, 39.1.0, 39.2.0, 40.0.0, 40.1.0, 40.1.1, 40.2.0, 40.3.0, 40.4.0, 40.4.1, 40.4.2, 40.4.3, 40.5.0, 40.6.0, 40.6.1, 40.6.2, 40.6.3, 40.7.0, 40.7.1, 40.7.2, 40.7.3, 40.8.0, 40.9.0, 41.0.0, 41.0.1, 41.1.0, 41.2.0, 41.3.0, 41.4.0, 41.5.0, 41.5.1, 41.6.0, 42.0.0, 42.0.1, 42.0.2, 43.0.0, 44.0.0, 44.1.0, 44.1.1, 45.0.0, 45.1.0, 45.2.0, 45.3.0, 46.0.0, 46.1.0, 46.1.1, 46.1.2, 46.1.3, 46.2.0, 46.3.0, 46.3.1, 46.4.0, 47.0.0, 47.1.0, 47.1.1, 47.2.0, 47.3.0, 47.3.1, 47.3.2, 48.0.0, 49.0.0, 49.0.1, 49.1.0, 49.1.1, 49.1.2, 49.1.3, 49.2.0, 49.2.1, 49.3.0, 49.3.1, 49.3.2, 49.4.0, 49.5.0, 49.6.0, 50.0.0, 50.0.1, 50.0.2, 50.0.3, 50.1.0, 50.2.0, 50.3.0, 50.3.1, 50.3.2, 51.0.0, 51.1.0, 51.1.0.post20201221, 51.1.1, 51.1.2, 51.2.0, 51.3.0, 51.3.1, 51.3.2, 51.3.3, 52.0.0, 53.0.0, 53.1.0, 54.0.0, 54.1.0, 54.1.1, 54.1.2, 54.1.3, 54.2.0, 56.0.0, 56.1.0, 56.2.0, 57.0.0, 57.1.0, 57.2.0, 57.3.0, 57.4.0, 57.5.0, 58.0.0, 58.0.1, 58.0.2, 58.0.3, 58.0.4, 58.1.0, 58.2.0, 58.3.0, 58.4.0, 58.5.0, 58.5.1, 58.5.2, 58.5.3, 59.0.1, 59.1.0, 59.1.1, 59.2.0, 59.3.0, 59.4.0, 59.5.0, 59.6.0)
ERROR: No matching distribution found for setuptools==68.0.0
```
I also see from the documentation that the recommended python versions are python 3.7-3.11, however i was only able to install python 3.6.8. This was the lastest version available when checked using "yum --showduplicates list python3". Regardless, I continued to proceed with the GCP integration and this was successful as shown further below:

#### Configuring GCP integration on the agent.

The same steps carried out on the server earlier were repeated here. I edited the agent's configuration file `/var/ossec/etc/ossec.conf` and moved the credentials.json file to the /var/ossec/wodles/gcloud/credentials.json location to configure GCP integration as follows:

```xml
 <gcp-pubsub>
     <pull_on_start>yes</pull_on_start>
     <interval>1m</interval>
     <project_id>beta-1-e2e-testing</project_id>
     <subscription_name>Beta1-E2E-Sub</subscription_name>
     <credentials_file>/var/ossec/wodles/gcloud/credentials.json</credentials_file>
 </gcp-pubsub>
```
![image](https://github.com/wazuh/wazuh/assets/85684507/cdb9fe9a-69ac-42b2-874f-a243e5896f27)

I did not proceed with testing bucket storage considering i can't create buckets

**Comment by Is3x** [comment]
### :red_circle: Cloud Security Posture Management ([BlogPost](https://wazuh.com/blog/cloud-security-posture-management/)) 

I did not get the alerts matching the particular rules configured in the blog post, however i received several "NOTICE" alerts for both use cases. Basically, I Restarted the GCP integration modules on the wazuh server but didn't get the configured alerting rules.

![ttrr](https://github.com/wazuh/wazuh/assets/85684507/62d591d2-deb3-40e2-b122-6d428ebada0e)

![ttr](https://github.com/wazuh/wazuh/assets/85684507/27a06158-709f-4cfb-ae10-bae46d4da02c)

**Comment by fdalmaup** [comment]
# Review

Great job, detailed and referencing previous known issues.  

LGTM!

**Comment by Is3x** [comment]
Thank you for your review @fdalmaup. I created an entirely new project for this test and there were no existing buckets due to this. I reached out to the CICD team and i was shown an existing project called **wazuh-dev** which i could use. I started testing from scratch in order to recreate everything, however for some reason, i was not able to view the logs from this project on my dashboard even though i rechecked this several times and i could see the log on the GCP web console. This missing prerequisite hindered me from using the buckets within this project to carry out this task hence i focused on the project i created but for which i was unable to create a bucket to perform the **Bucket storage test** case
